### PR TITLE
Improvements to setup scripts, library functions

### PR DESCRIPTION
**Pull request by crogers1**
and implements two additional testcases.

[autolaunch.py]
  * Bug fixes, better error checking/handling

[bvt.py]
  * Improved checking for *.py files
  * New cmd line arg to set timelimit for testcases that require it.

[bvt_daemon.py]
  * Exception handling for ValueError, seen on some return codes.

[docs/README.md]
  * Section on setting up ddns and masquerading
  * more details on windows vm configuration with execdaemon
  * Notes that we support AMT v9.0 over the wsman protocol

[setup/00-install-prereqs.sh]
  * Fixed package version problem
  * added fetch for amtctrl tool used for wsman

[setup/03-install-dhcp-serv.sh]
  * Added optional dnssec-tools and bind9 pkgs for ddns

[src/bvtlib/call_exec_daemon.py]
  * Modified output to return similar to run.py, options to split
    on word or line.

[src/bvtlib/device_ops.py]
  * New module that performs operations on cd devices.

[src/bvtlib/disk_ops.py]
  * New module to perform operations on disks such as
    snapshotting, add/removing vhds, retrieve vhd name, is vhd valid, etc.

[src/bvtlib/guest_info.py]
  * Added new functions to query information about the guest
    (ram, stubdom, domid, vcpus, uuid, tools, etc)

[src/bvtlib/guest_ops.py]
  * Added new functions to support some of the new testcases.
  * Bug fixes in create_guest(), wrong param name

[src/bvtlib/host_ops.py]
  * New module to perform operations at the host level such as:
	mount debugfs, downloading vhds, reboot. New host functions should
	go in here as well.

[src/bvtlib/install_tools_windows.py]
  * Removed hard-coded username for windows vm, replaced with function
	to query the user.

[src/bvtlib/power_control.py]
  * Added wsman support for AMT v9.0+. Power control type can be set
	in mongodb for a DUT.

[src/bvtlib/record_test.py]
  * Slightly increase timeout to avoid exceptions.

[src/bvtlib/run.py]
  * Disable strict host key checking and known host file recording to
	make sshing to DUTs or bvt-controller more transparent

[src/bvtlib/windows_transitions.py]
  * Fix import so module name isn't required in function call
  * Added functions to support sleep_test testcase

[src/bvtlib/xen_info.py]
  * New module to query information from xenops

[src/testcases/cd_exclusive_test.py]
  * New testcase to ensure that cd devices can only be passed through
	to one guest at a time.

[src/testcases/sleep_test.py]
  * New testcase to verify that s3 works for guest and certain host operations
	function properly while a guest is in s3.

Signed-off-by: Chris Rogers <rogersc@ainfosec.com>

### CODE REVIEW

**Comment by crogers1** [comment]
The majority of this PR adds a lot of new functions to the bvtlib modules to both support new testcases (several of which I am unable to upstream at this time, but will hopefully in the future) and to provide more functionality.  I'm happy to clarify anything that needs explaining.

**Comment by rossphilipson** [comment]
We are going to leave this open for a while until we can get some bvt kit up and running at our office to test things like this. We currently have no knowledge or experience with bvt. We have a plan, Chris is going to come here and show us the path...

**Comment by jean-edouard** [comment]
With help from @crogers1, I set up a BVT controller with a few test nodes.
We used the code from this PR for the whole process.

I ran into a bunch of issues, see https://openxt.atlassian.net/browse/OXT-550

Some of these issues are documentation-related and can fixed in a later PR.

However, the technical issues I ran into should probably be fixed.
For each issue, it's unclear if it is caused by this PR or not, since I never tested BVT before.

Is there any chance you could fix the technical issues by adding one or more commits to this PR?

**Comment by jean-edouard** [comment]
TL;DR: Merging soon
Without this commit, BVT is pretty much unusable...
I found bugs when testing this PR (see previous comment), that may or may not be related to the actual changes.
The original author doesn't have much spare time, and more people will be getting involved with BVT in the near future, so I think it would be best to have this PR in.
I therefore suggest we merge this, and I will do it tomorrow if there's no objection.
